src/main/java/weizberg/citibike/lambda/LambdaServiceFactory.java

```diff
@@ -23,9 +24,8 @@ public  LambdaServiceFactory() {
         this.lambdaService = retrofit.create(LambdaService.class);
     }
 
-    public CitibikeResponse callLambda(CitibikeRequest request) throws IOException {
-        Call<CitibikeResponse> call = lambdaService.callLambda(request);
-        return call.execute().body();
+    public Single<CitibikeResponse> callLambda(CitibikeRequest request) throws IOException {
+        return lambdaService.callLambda(request);
     }
```

Comment:
This doesn't really belong in the Factory.
The factory is just for creating LambdaService objects.